src/checkout/utils/checkout.ts

```diff
@@ -0,0 +1,63 @@
+// import * as yup from 'yup'
+import {BackendContact, Contact, isUS} from 'src/checkout/utils/contact'
+import {BillingNotifySettings} from 'src/types/billing'
+
+const requiredMessage = 'This is a required field'
+
+export const minimumBalanceThreshold = 1
+
+// export const notificationSettingsSchema: yup.ObjectSchema<NotificationSettings> = yup.object(
+//   {
+//     shouldNotify: yup.boolean().required(requiredMessage),
+//     notifyEmail: yup.string().required(requiredMessage),
+//     balanceThreshold: yup
+//       .number()
+//       .min(minimumBalanceThreshold, 'Please enter a value of ${min} or greater')
+//       .required(requiredMessage)
+//       .transform(value => (isNaN(value) ? undefined : value)),
+//   }
+// )
+
+// export const contactSchema: yup.ObjectSchema<Contact> = yup.object({
+//   street1: yup.string(),
+//   street2: yup.string(),
+//   city: yup.string().required(requiredMessage),
+//   country: yup.string().required(requiredMessage),
+//   intlSubdivision: yup.string().notRequired(),
+//   usSubdivision: yup.string().when('country', {
+//     is: isUS,
+//     then: yup.string().required(requiredMessage),
+//     otherwise: yup.string().notRequired(),
+//   }),
+//   postalCode: yup.string().when('country', {
+//     is: isUS,
+//     then: yup.string().required(requiredMessage),
+//     otherwise: yup.string().notRequired(),
+//   }),
+// })
+
+interface CheckoutBase {
+  paymentMethodId?: string
+}
+
+export type Checkout = CheckoutBase & BillingNotifySettings & Contact
+
+export type BackendCheckout = CheckoutBase & BackendContact
+
+// export const validationSchema: yup.ObjectSchema<Checkout> = yup
+//   .object({
+//     paymentMethodId: yup.string().nullable(),
+//   })
+//   .concat(notificationSettingsSchema)
+//   .concat(contactSchema)
+
+export const toBackend = (checkout: Checkout): BackendCheckout => {
+  const {intlSubdivision, usSubdivision, ...values} = checkout
+
+  const subdivision = isUS(values.country) ? usSubdivision : intlSubdivision
+
+  return {
+    ...values,
+    subdivision,
+  }
+}
```

Comment:
we can probably delete this file and localize anything that's using what's in it (if it's only in 1 place, which i imagine it is)